lodash.js

```diff
@@ -7496,6 +7496,29 @@
       return length ? baseSlice(array, 0, -1) : [];
     }
 
+    /**
+     * Inserts `item` at the given `index` into the `array`.
+     *
+     * @static
+     * @memberOf _
+     * @since 4.7.15
+     * @category Array
+     * @param {number} [index=0] The desired index of the inserted the item.
+     * @param {*} item The item to be inserted.
+     * @param {Array} array The array which will have an item inserted.
+     * @returns {Array} Returns a new array with the item inserted at the given index.
+     * @example
+     *
+     * _.insert(2, 'Hello', ['Hey', 'Hi', 'Howdy'])
+     * // => '['Hey', 'Hi', 'Hello', 'Howdy']'
+     * 
+     * _.insert(-2, 'Hello', ['Hey', 'Hi', 'Howdy'])
+     * // => '["Hey", "Hello", "Hi", "Howdy"]'
+     */
+    function insert(index = 0, item, array = []) {
+      return [...array.slice(0, index), item, ...array.slice(index)];
+    }
```

Comment:
We do data first in our generic methods so it would be `array`, `index`, `value`.

Reply:
Shouldn't it be `array`, `value`, `index` for cases where `index` is omitted and `value` should be inserted to default `index` (0?)

Reply:
Thanks. Sorry for taking so long but I've finally resolved this.